snapshot-labs/snapshot-sidekick
# chore: uncouple the cache engine from votesReport

## 🧿 Current issues / What's wrong ?

The votesReport class has the cache handling hardcoded into it.

When adding another class using the cache, we will have to duplicate the cache logic

## 💊 Fixes / Solution

Uncouple the cache logic from the votesReport class

## 🚧 Changes

- Extract the cache logic into a `Cache` class. We can now test the cache separately
- `votesReport` is now inheriting the `Cache` class
- the `queue` function from the Queue lib now takes a Cache object as argument
- the webhook api endpoint has been refactored to handle other types of events
- the cache support `Buffer` type on `set` and `get`

## 🛠️ Tests

- choose a proposal with few votes
- `curl -X POST localhost:3005/api/votes/[PROPOSAL-ID]`
- it should generate a cache file
- call the curl command again
- it should return the cache CSV content
